src/Rings/walk.jl

```diff
@@ -0,0 +1,1724 @@
+	end
+end
+
+# returns the next t to compute the next weight vector w(t) = w + t * (tw - w) like it´s done in Cox, Little & O'Sheao (2005).
+function next_weight_fractal_walk(
+	G::Oscar.IdealGens,
+	cweight::Array{T, 1},
+	tweight::Array{K, 1},
+) where {T <: Number, K <: Number}
+	if (cweight == tweight)
+		return [0]
+	end
+	tmin = 1
+	for v in difference_lead_tail(G)
+		tw = dot(tweight, v)
+		if tw < 0
+			cw = dot(cweight, v)
+			t = cw // (cw - tw)
+			if t < tmin
+				tmin = t
+			end
+		end
+	end
+
+	# BigInt is needed to prevent overflows in the conversion of the weight vectors.
+	return BigInt(numerator(tmin)) // BigInt(denominator(tmin))
+end
+
+# returns 'true' if the leading terms of G w.r.t the matrixordering T are the same as the leading terms of G w.r.t the weighted monomial ordering with weight vector of pvecs[p] (pvecs[p-1]) and the matrixordering T.
+function inCone(
+	G::Oscar.IdealGens,
+	T::Matrix{Int},
+	pvecs::Vector{Vector{Int}},
+	p::Int,
+)
+	# pvecs(1) equals the first row of T
+	if p == 1
+		return true
+	end
+	ord = matrix_ordering(base_ring(G), T)
+	cvzip = zip(
+		gens(G),
+		initials(G, pvecs[p-1]),
+		initials(G, pvecs[p]),
+	)
+	for (g, in, in2) in cvzip
+		if !isequal(
+			leading_term(g, ordering = ord),
+			leading_term(in, ordering = ord),
+		) ||
+		   !isequal(
+			leading_term(g, ordering = ord),
+			leading_term(in2, ordering = ord),
+		)
+			return false
+		end
+	end
+	return true
+end
+
+
+#################################################################
+# Procedures of the generic walk.
+# The generic walk is proposed by Fukuda, Lauritzen & Thomas (2005).
+#################################################################
+
+# returns the initials of the polynomials w.r.t. the vector v.
+function facet_initials(
+	G::Oscar.IdealGens,
+	lm::Vector{T},
+	v::Vector{Int},
+) where {T <: MPolyRingElem}
+	Rn = parent(first(G))
+	initials = Array{Singular.elem_type(Rn), 1}(undef, 0)
+	count = 1
+	for g in G
+		inw = Singular.MPolyBuildCtx(Rn)
+		el = first(Singular.exponent_vectors(lm[count]))
+		for (e, c) in
+			zip(Singular.exponent_vectors(g), Singular.coefficients(g))
+			if el == e || isparallel(el - e, v)
+				Singular.push_term!(inw, c, e)
+			end
+		end
+		h = finish(inw)
+		push!(initials, h)
+		count += 1
+	end
+	return initials
+end
+
+# returns the differences of the exponent vectors of the leading terms and the polynomials of the generators of I.
+function difference_lead_tail(
+	G::Oscar.IdealGens,
+	Lm::Vector{L},
+	T::Union{Matrix{N}, MatElem{N}},
+) where {L <: MPolyRingElem, N}
+	v = Vector{Int}[]
+	for i ∈ 1:length(G)
+		ltu = collect(AbstractAlgebra.exponent_vectors(leading_term(Lm[i], ordering = matrix_ordering(base_ring(G), T))))[1]
+		for e in AbstractAlgebra.exponent_vectors(G[i])
+			if ltu != e
+				push!(v, ltu .- e)
+			end
+		end
+	end
+	return unique!(v)
+end
+
+# returns true if the vector u is parallel to the vector v.
+function isparallel(u::Vector{Int}, v::Vector{Int})
```

Comment:
I would recommend spelling out what you mean by "parallel" -- here it seems that `u` must be a multiple of `v` (except that if `u` is a zero vector, but `v` is not, in your code they are not parallel; while if `v` is a zero vector and `u` is not, they *are* parallel...?!? Maybe `u` and `v` are never supposed to be zero vectors? Then state so. Also, I guess you expect them to have equal length, so perhaps add an `@assert that verifies this..)